### ISSUE TITLE
[switch-exhaustiveness-check] Autocorrect incorrect parsing/handling property names with hyphens in them

### ISSUE THREAD

**Issue by jpike88** [typescript-eslint/typescript-eslint#2181]
**Repro**

Just have an enum with at least one property containing a hyphen:

```TS
export enum MyEnum {
	'blah-blah' = 'blah-blah',
	'test' = 'test'
}
```

```JSON
{
  "rules": {
    "@typescript-eslint/switch-exhaustiveness-check": "error"
  }
}
```

```TS
function myFunction(
		arg: MyEnum
	): string {
		switch (reason) {

		}
	}

```

**Expected Result**

```TS
function myFunction(
		arg: MyEnum
	): string {
		switch (reason) {
case MyEnum['blah-blah']: { throw new Error('Not implemented yet: MyEnum.['blah-blah'] case') }
case MyEnum.test: { throw new Error('Not implemented yet: MyEnum.test case') }		}
	}

```

**Actual Result**

```TS
function myFunction(
		arg: MyEnum
	): string {
		switch (reason) {
case MyEnum.blah-blah: { throw new Error('Not implemented yet: MyEnum.blah-blah case') }
case MyEnum.test: { throw new Error('Not implemented yet: MyEnum.test case') }		}
	}

```
**Additional Info**

As you can see the above code will error, as a property name containing hyphens hasn't been put in quotes.

On latest version of typescript@eslint-plugin 2.33.0
